src/abstractitensornetwork.jl

```diff
@@ -585,36 +585,58 @@ end
 
 # For ambiguity error; TODO: decide whether to use graph mutating methods when resulting graph is unchanged?
 function _orthogonalize_edge(tn::AbstractITensorNetwork, edge::AbstractEdge; kwargs...)
+  return _orthogonalize_edges(tn, [edge]; kwargs...)
+end
+
+# For ambiguity error; TODO: decide whether to use graph mutating methods when resulting graph is unchanged?
+function _orthogonalize_edges(
+  tn::AbstractITensorNetwork, edges::Vector{<:AbstractEdge}; kwargs...
+)
   # tn = factorize(tn, edge; kwargs...)
   # # TODO: Implement as `only(common_neighbors(tn, src(edge), dst(edge)))`
   # new_vertex = only(neighbors(tn, src(edge)) ∩ neighbors(tn, dst(edge)))
   # return contract(tn, new_vertex => dst(edge))
   tn = copy(tn)
-  left_inds = uniqueinds(tn, edge)
-  ltags = tags(tn, edge)
-  X, Y = factorize(tn[src(edge)], left_inds; tags=ltags, ortho="left", kwargs...)
-  tn[src(edge)] = X
-  tn[dst(edge)] *= Y
+  for edge in edges
+    left_inds = uniqueinds(tn, edge)
+    ltags = tags(tn, edge)
+    X, Y = factorize(tn[src(edge)], left_inds; tags=ltags, ortho="left", kwargs...)
+    tn[src(edge)] = X
+    tn[dst(edge)] *= Y
+  end
   return tn
 end
 
 function ITensorMPS.orthogonalize(tn::AbstractITensorNetwork, edge::AbstractEdge; kwargs...)
   return _orthogonalize_edge(tn, edge; kwargs...)
 end
 
+function ITensorMPS.orthogonalize(
```

Comment:
I think it would be better to rename the versions that orthogonalize along an edge or sequence of edges to something else (I know I made the name originally so that was my fault).

There are two distinct concepts, orthogonalizing _along_ a path/trail/walk (https://en.wikipedia.org/wiki/Path_(graph_theory)), and orthogonalizing _towards_ a region. My concern with the current naming scheme is that either one could use vertices or edges, but are really distinct functionality, so it isn't great from a generic coding perspective. What if we want to define the region we orthogonalize towards by a set of edges, or the path we want to orthogonalize along by a set of vertices?

We use `orthogonalize` to mean orthogonalizing towards a region in `ITensorMPS.jl`, and I think that will be the more common operation that external users will use, so I think we should stick with that name. So then I think we should rename this function that defines a path/trail/walk to orthogonalize along as something else. It could be `orthogonalize_walk` based on the terminology here: https://en.wikipedia.org/wiki/Path_(graph_theory). An alternative is `orthogonalize_trail` but I could imagine cases where you want to repeat edges, say on a loopy graph, so maybe it is better to be more general. We could also call it `orthogonalize_along` but I slightly prefer using graph terminology.

Reply:
Yeah that's a good point that we are using the same terminology for two somewhat distinct concepts. What about `orthogonalize_sequence`?